### PR TITLE
Mat 5925 import test cases no overwriting

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-5925](https://jira.cms.gov/browse/MAT-5925)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java [hunk 1]
```diff
@@ -375,21 +393,164 @@ public List<TestCaseImportOutcome> importTestCases(
...
+      log.info(
+          "User {} is unable to import test case with patient id : "
+              + "{} because of JsonProcessingException: "
+              + ex.getMessage(),
+          userName,
+          testCaseImportRequest.getPatientId());
+      return TestCaseImportOutcome.builder()
+          .patientId(testCaseImportRequest.getPatientId())
+          .successful(false)
+          .message("Error occurred")
```
Comment (RohitKandimalla) [inline src/main/java/cms/gov/madie/measure/services/TestCaseService.java]: Can this error be more specific? There are errors that are displayed to the user, So a brief explanation could help them to debug and fix them before reaching out to HD.

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java [hunk 2]
```diff
@@ -99,4 +107,95 @@ public static boolean isUuid(String value) {
...
+          }
+        }
+      }
+    }
+
+    return null;
+  }
+
+  public static List<TestCaseGroupPopulation> getTestCaseGroupPopulationsFromMeasureReport(
+      String json) throws JsonProcessingException {
```
Comment (RohitKandimalla) [inline src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:146]: Wondering if we can use Hl7.FHIR.r4.models to convert TestCaseJson to a FHIR Object and fetch all necessary information without parsing through each node.

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java]: @RohitKandimalla Sure will do.

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:146]: @RohitKandimalla That's a good idea, will check the model to use, thanks a lot for the suggestion!

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:146]: Talked to @RohitKandimalla, due to the release date constraint, this change will be left to future implementation.

**Review by RohitKandimalla — APPROVED** [review]

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java]: This is done so will resolve it.

**Pull request merged**